src/map/npc.cpp

```diff
@@ -900,16 +913,28 @@ int npc_event(struct map_session_data* sd, const char* eventname, int ontouch)
 
 	switch(ontouch)
```

Comment:
I would like to suggest that you replace the hole switch statement with an `if/elseif` statement since it look more clear. Also you were discarding the return value of `npc_event_sub(sd,ev,eventname)`, when before it wasn't.
```suggestion
if (ontouch == 1) {
    if (pc_ishiding(sd))
        return 0;

    nd->touching_id = sd->bl.id;

    auto it = std::find(sd->npc_ontouch_.begin(), sd->npc_ontouch_.end(), nd->bl.id);
    if (it == sd->npc_ontouch_.end())
        sd->npc_ontouch_.push_back(nd->bl.id);

} else if (ontouch == 2) {
    auto it = std::find(sd->areanpc.begin(), sd->areanpc.end(), nd->bl.id);
    if (it == sd->areanpc.end())
        sd->areanpc.push_back(nd->bl.id);
}

return npc_event_sub(sd,ev,eventname);
```